dashboard/http_server_head.py

```diff
@@ -145,6 +182,39 @@ async def run(self, modules):
             logger.info(r)
         logger.info("Registered %s routes.", len(dump_routes))
 
+    def _setup_metrics(self, app):
+        metrics = DashboardPrometheusMetrics()
+        app["metrics"] = metrics
+        app.middlewares.insert(0, self.metrics_middleware)
```

Comment:
Just OOC, is there a reason why we declare middleware here instead of following https://us-pycon-2019-tutorial.readthedocs.io/aiohttp_middlewares.html? (create a middleware with a decorator and add it when `app = aiohttp.web.Application(client_max_size=100 * 1024 ** 2)` is called)

Reply:
let me refactor to be in that format.